tools/zim_install

```diff
@@ -0,0 +1,158 @@
+main() {
+  # Use colors, but only if connected to a terminal, and that terminal
+  # supports them.
+  if which tput >/dev/null 2>&1; then
+    ncolors=$(tput colors)
+  fi
+  if [ -t 1 ] && [ -n "${ncolors}" ] && [ "${ncolors}" -ge 8 ]; then
+    red="$(tput setaf 1)"
+    green="$(tput setaf 2)"
+    yellow="$(tput setaf 3)"
+    blue="$(tput setaf 4)"
+    bold="$(tput bold)"
+    normal="$(tput sgr0)"
+  else
+    red=""
+    green=""
+    yellow=""
+    blue=""
+    bold=""
+    normal=""
+  fi
+
+  # Only enable exit-on-error after the non-critical colorization stuff,
+  # which may fail on systems lacking tput or terminfo
+  set -e
+
+  # Yes or no function
+  yesno() {
+    printf "${*} "
+    read answer
+    if printf "${answer}\n" | grep -iq "^y"; then
+        return 0
+    else
+        return 1
+    fi;
+  }
+
+  # Check if `zsh` exists, or if `zsh` isn't too old
+  if [ ! "$(grep "/zsh$" "/etc/shells")" ] || \
+    env zsh -c "autoload -Uz 'is-at-least'; ! is-at-least 5.2"; then
```

Comment:
Shouldn't we use `&&` instead of the first `;`?

Reply:
How do you figure that? It's a basic `if x | y; then ..`